linera-core/src/worker.rs

```diff
@@ -766,15 +723,21 @@ where
     pub async fn handle_certificate(
         &self,
         certificate: Certificate,
-        hashed_certificate_values: Vec<HashedCertificateValue>,
         blobs: Vec<Blob>,
         notify_when_messages_are_delivered: Option<oneshot::Sender<()>>,
     ) -> Result<(ChainInfoResponse, NetworkActions), WorkerError> {
         trace!("{} <-- {:?}", self.nickname, certificate);
+        let bytecode_blobs = blobs
+            .iter()
+            .filter(|blob| {
+                blob.id().blob_type == BlobType::ContractBytecode
+                    || blob.id().blob_type == BlobType::ServiceBytecode
+            })
+            .collect::<Vec<_>>();
         ensure!(
-            certificate.value().is_confirmed() || hashed_certificate_values.is_empty(),
-            WorkerError::UnneededValue {
-                value_hash: hashed_certificate_values[0].hash(),
+            certificate.value().is_confirmed() || bytecode_blobs.is_empty(),
+            WorkerError::UnneededBlob {
+                blob_id: bytecode_blobs[0].id()
             }
```

Comment:
We shouldn't make any distinction between bytecode and other blobs here anymore.

I think the idea here was that if we receive a `ValidatedBlock` certificate, we won't do anything with its content: No execution, no further validation, just a confirmation vote. So we wouldn't need any certificate values at this point.

For blobs, we now _do_ seem to check that they are complete inside `process_validated_block`, but then we don't save them! We should either:
1. do this check here for all blobs, and not pass any blobs into `process_validated_block`; or:
2. remove this check and keep the blobs in the chain manager.

I think for liveness, this would mean:
1. A validator's chain manager always contains all blobs that belong to any `ValidatedBlock` _vote_ this validator has signed in any round no earlier than the locked block.
2. A validator's chain manager always contains all blobs that belong to the locked block.

2 actually seems simpler. But we probably still want to _cache_ the blobs when we handle a blob proposal, so that in most cases we don't have to send them again when sending the `ValidatedBlock` certificate.